src/kagglehub/models_helpers.py

```diff
@@ -30,8 +38,18 @@ def _create_model_instance(model_handle: ModelHandle, files: List[str], license_
     logger.info(f"Your model instance has been created.\nFiles are being processed...\nSee at: {model_handle.to_url()}")
 
 
-def _create_model_instance_version(model_handle: ModelHandle, files: List[str], version_notes: str = "") -> None:
-    data = {"versionNotes": version_notes, "files": [{"token": file_token} for file_token in files]}
+def _create_model_instance_version(
+    model_handle: ModelHandle, files_and_directories: UploadDirectoryInfo, version_notes: str = ""
+) -> None:
+    serialized_data = [
```

Comment:
Can you move this logic to the `UploadDirectoryInfo` class? Add a `def serialize(self): ...` to it and call it here.